### ISSUE TITLE
Random Signer confirm/reject error pop-up

### ISSUE THREAD

**Issue by jacogr** [ethcore/parity#3570]
![parity 2016-11-22 13-25-09](https://cloud.githubusercontent.com/assets/1424473/20526062/b0790c3c-b0c2-11e6-9a57-2dcefb2498c4.png)
![parity 2016-11-22 13-23-59](https://cloud.githubusercontent.com/assets/1424473/20526063/b07b867e-b0c2-11e6-88ba-385d5a78d91e.png)

**Comment by jacogr** [comment]
_could_ be from finding the request in the pending queue in the reducer -

```
signerSuccessConfirmRequest (state, action) {
    const { id, txHash } = action.payload;
    const confirmed = Object.assign(
      state.pending.find(p => p.id === id),
      { result: txHash, status: 'confirmed' }
    );

    return {
      ...state,
      pending: removeWithId(state.pending, id),
      finished: [confirmed].concat(state.finished)
    };
  }
```

i.e. `state.pending.find` - really no proper insight at this time

**Comment by ngotchac** [comment]
Should be fixed from #3804

**Comment by ngotchac** [comment]
Should definitely be fixed with #3990
